### ISSUE TITLE
Allow complex database queries in plugins

### ISSUE THREAD

**Issue by raffael** [baasbox/baasbox#651]
Plugins are awesome!

It's great how easy it is to fetch simple database rows using ```Box.Documents.find()```.
In some situations however we need more complex queries, to e.g., calculate a highscore list of all users.

Exposing a generic ```Box.Documents.query("SELECT ...")``` like API would be great. Is this planned for the future?

**Comment by ffnoname** [comment]
+1 for that. I am in desperate need of a Box.Documents.Count("collection", params) :)
could be achieved by mentioned generic Box.Documents.query("SELECT count ....");
Do you have an idea for a workaround? I am currently using Box.Documents.find().length but thats obviously way too slow.

**Comment by giastfader** [comment]
You can try with:
```
var ret=Box.Documents.find("collection",{
            fields:"...",
            where:"...'",
            orderBy:"...",
            count:true
    })
```

**Comment by raffael** [comment]
In this simple case I'd personally make use of
```
    fields: "COUNT(id) AS count",
    [...]
```

However, there are still situation where a direct access to the OrientDB query interface would be required.

**Comment by ffnoname** [comment]
cool. thanks for the tips! I still +1 the Box.Documents.query(..). my current problem is solved but I also find it useful to have direct access to OrientDB query interface.

**Comment by dirkam** [comment]
+1

Any plans on having this feature added in the near future?

**Comment by avently** [comment]
For some reason ordering by distance() doesn't work if not to write LET keyword in sql query. Can you give us method to use something like: 

Box.Users.find({fields: "*, $distance", let : "$distance = distance(lat, long, 123.0, 321.0)", where: "$distance <= 1000", orderBy : "$distance asc"});

**Comment by giastfader** [comment]
closed by #729

**Comment by giastfader** [comment]
https://github.com/giastfader/baasbox/commit/08a7bbb62a1e8cdbf29938cac86b1f6186d4d926

and

https://github.com/giastfader/baasbox/commit/52e6fa3e35767226613ffdb6612f6ef7ac8cbf91
